### PR TITLE
AAG-2774: iOS: UIWindow Removal

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2774]

## DESCRIPTION
This PR modifies how the Parent Gate and Bumper screens are presented, instead of using a UIWindow, they are now presented on the top most view controller. This implementation enables these features to work with SwiftUI.

## SCREENSHOTS

Interstitial / Banner

https://user-images.githubusercontent.com/618350/214373836-d9472a50-530c-4dec-b275-a11683da259f.mov

Video: 

https://user-images.githubusercontent.com/618350/214599507-128455f4-9824-4fdd-867e-1904b3bb704a.mov

[AAG-2774]: https://superawesomeltd.atlassian.net/browse/AAG-2774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Pod/Classes/UI/Common/ParentalGate.swift [hunk 1]
```diff
@@ -83,7 +82,9 @@ class ParentalGate {
 
         if let controller = errorAlertController {
             controller.addAction(okAction)
-            topWindow = controller.presentInNewWindow()
+            controller.modalTransitionStyle = .crossDissolve
```
Comment (thomas-james-1986) [inline Pod/Classes/UI/Common/ParentalGate.swift:85]: I wonder if there are other presentation styles that would ensure the view displays over content in other apps such as `.overCurrentContext`?

Comment (me2k) [reply Pod/Classes/UI/Common/ParentalGate.swift:85]: That would indeed work, I'll add it. 👍

File: Example/Podfile [hunk 2]
```diff
@@ -7,15 +7,15 @@ use_frameworks!
 target 'SuperAwesomeExample' do
   pod 'SuperAwesome', :path => '../'
 
-  pod 'Firebase/AnalyticsWithoutAdIdSupport'
-  pod 'FirebaseDatabaseSwift'
+  pod 'Firebase/AnalyticsWithoutAdIdSupport', '~> 9.6.0'
+  pod 'FirebaseDatabaseSwift', '~> 9.6.0'
```
Comment (gunhansancar) [inline Example/Podfile:11]: Nice 👍

File: Pod/Classes/UI/Common/BumperPage.swift [hunk 3]
```diff
@@ -202,7 +200,8 @@ public class BumperPage: UIViewController, Injectable {
     func play(_ onComplete: VoidBlock?) {
         self.onComplete = onComplete
         self.modalPresentationStyle = .overCurrentContext
-        topWindow = presentInNewWindow()
+        self.modalTransitionStyle = .crossDissolve
```
Comment (thomas-james-1986) [inline Pod/Classes/UI/Common/BumperPage.swift:203]: I wonder if this one should use `modalPresentationStyle = .overCurrentContext` as well?

**Review by gunhansancar — APPROVED** [review]

**Review by thomas-james-1986 — APPROVED** [review]
👍

**Pull request merged**